listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -695,6 +695,36 @@ def delete(self, musicbrainz_id):
             self.log.error("Cannot delete listens for user: %s" % str(e))
             raise
 
+    def delete_listen(self, musicbrainz_id:str, recording_msid: str):
+        """ Delete a particular listen for user with specified MusicBrainz ID.
+
+        Args:
+            musicbrainz_id: the MusicBrainz ID of the user
+            recording_msid: the MessyBrainz ID of the recording
+
+        Raises: Exception if unable to delete the listen
+        """
+
+        args = {'user_name': musicbrainz_id, 'recording_msid': recording_msid}
+        query = """DELETE FROM listen
+                    WHERE user_name = :user_name
+                      AND recording_msid = :recording_msid"""
+
+        try:
+            with timescale.engine.connect() as connection:
+                connection.execute(sqlalchemy.text(query), args)
+
+            listen_count = cache.get(self.ns + TimescaleListenStore.REDIS_TIMESCALE_TOTAL_LISTEN_COUNT, decode=False)
+            if listen_count:
+                cache.set(
+                    self.ns + TimescaleListenStore.REDIS_TIMESCALE_TOTAL_LISTEN_COUNT,
+                    int(listen_count) - 1,
+                    TimescaleListenStore.TOTAL_LISTEN_COUNT_CACHE_TIME,
```

Comment:
Perhaps this not ideal, perhaps it doesn't matter. But if the listen count just near expiry and we delete a listen and set the decremented listen count for the same duration again the date may be stale for longer. How about just deleting the key from redis and forcing a fetch next time?

Reply:
From your suggestion, I think we can just delete the key and also decrement the count at frontend on success. If the user reloads the page then the new listen_count will be updated in the cache also. Sounds good to you?

Reply:
OK, that works for me.